File: backend/RMarenco.FinalProject.NorthWindTraders/Core/NorthWindTraders.Domain/Entities/Order.cs
using System.Diagnostics.CodeAnalysis;
using System.Text.Json.Serialization;

namespace NorthWindTraders.Domain.Entities
{
    public class Order
    {
        public int OrderID { get; set; }
        public string CustomerID { get; set; }
        public int EmployeeID { get; set; }
        public DateTime OrderDate { get; set; }
        public int ShipVia { get; set; }
        public string ShipAddress { get; set; }
        public string ShipCity { get; set; }
        public string ShipRegion { get; set; }
        public string ShipPostalCode { get; set; }
        public string ShipCountry { get; set; }
        public Customer Customer { get; set; }
        public Employee Employee { get; set; }
        public Shipper Shipper { get; set; }
    }
}

File: frontend/northwind-traders/src/types/Order/order.ts
import { ICustomer } from "../Customer/customer";
import { IEmployee } from "../Employee/employee";
import { IOrderDetail } from "../OrderDetail/orderDetail";
import { IShipper } from "../Shipper/shipper";

export interface IOrder {
  orderID: number;
  orderDate: Date;
  shipAddress: string;
  shipCity: string;
  shipRegion: string;
  shipPostalCode: string;
  shipCountry: string;
  customer?: ICustomer;
  shipper?: IShipper
  employee?: IEmployee;
}

export interface IOrderWithDetails {
  order: IOrder;
  orderDetails: IOrderDetail[];
}